### ISSUE TITLE
Wrong support selected.

### ISSUE THREAD

**Issue by smaugsmog** [MathewSachin/Fate-Grand-Automata#89]
This happened with both older versions and the newest one.

**Describe the bug**
Very rarely, a tamamo with mona lisa is selected instead of the preferred Merlin with a mona lisa. I have nothing set as the defaults (Both Servant and CE are set as any) and merlin and mlb mona lisa selected for the specific autoskill that I am using. This is also not always selecting tamamos with monalisas. I have watched it scroll past them before no problem. It is also very uncommon. I think it has been 4 or 5 times out of maybe 70-100 runs with merlin. I don't have a recording of it happening, since screen recording software seems to conflict with this app, but it happened once when I had just looked at my phone, so there shouldn't have been any notifications or popups that might have caused interference.

It happens so infrequently it is kind of hard to test. It's not too big of a deal either, but figured I should submit a report anyway. Thanks for working on this

**Device Info (please fill atleast 'Device model'):**
 - Device model: Google Pixel XL
 - Android Version: 10

**Comment by MathewSachin** [comment]
Thank you for opening the issue. I had also set Merlin+Mona Lisa as preferred but Tamamo+Mona Lisa selected atleast 3 times out of many runs.
I'm already looking into the problem.

**Comment by tchofy** [comment]
Not much to add here, as the developer probably already figured out what's happening, but for other users running into this issue:

Looks like the Support option is overriding the Autoskill support option. For a temporary fix, just go into the Support option on the main screen and set it the same as you have in your current autoskill script.

**Comment by MathewSachin** [comment]
Autoskill support options override Support options. So, if you have not set preferred Servant/CE in autoskill options, the ones set in support options are used. This is similar to how FGO-Lua behaves.

**Comment by tchofy** [comment]
Indeed, but that's the issue I had. I had a script set to pick waver with monalisa in autoskill, while the Support option had any servant with caldea lunchtime. The first few times the script would work as intended, but then suddenly it picked a lunchtime scherezade. I interrupted it before it could start, restarted the app, and it picked waver again, but then it kept scrolling while there were multiple wavers on-screen.
I did some tests to try to figure why it was happening, like setting the autoskill support for Merlin with any ce, then any support with monalisa, and it would still keep looking for something else, until I thought about trying to change the support on the home screen option.

I'll try to record the screen if it happens again and figure reproduction steps, I know how important it is to be able to reproduce a bug to be able to fix it.
